### ISSUE TITLE
initialization of K and L matrices

### ISSUE THREAD

**Issue by lelikchern** [HaberGroup/SemiImplicitDNNs#1]
Hi! Thank you for this work, it is very interesting and elegant. 
I am trying to apply IMEXnet for image denoising, in my case the noise is quite position-dependent, so I though that a quick enlargement of the receptive field might help.
Could you share some insights as to how to initialize the K and L matrices?
Namely, I see you have random numbers for initialization, and sometimes you multiply them by 1e-3, sometimes not. What would be the rule of thumb for initializitation - how would the amplitude of weights in L differ from those of K?
Did you try standard inits (e.g. xavier, kaiming) for K or L, or it does not make sense?

Another question related to _h_. Does changing its value generally result in quicker convergence?

**Comment by klensink** [comment]
Hi there, thanks for your interest in our work!

At the end of the day you'll need to atleast initialize **K**, **L** so that the network atleast starts out stable for your choice of *h*, but you may want to choose them to get certain behaviour. 

The weights in **K** are the convolution weights of the non-linear term that are treated explictely, so you could think of initializing them no different than you would if you were training a ResNet (where usually h=1) (e.g. Xavier as you mentioned). 

The weights in **L** are the convolution weights of the linear implicit term, so the size of these weights determine how much implicit coupling will occur. In eqn 6 [of the paper](https://arxiv.org/pdf/1903.02639.pdf) you can see that if **L=0** we are left with just an identity matrix out front, and the h**LY** term goes to zero so we return to just a ResNet (eqn 1). 

How you initialize **L** will influence the amount of implicit coupling as well as the stability of the forward propogation. For example in Figure 1, the difference between the middle row has **L=0** while the bottom row has **L** as defined in eqn 7. Additionally as discussed in section 2.3 the choice of L will effect the stability of the network, as changing the scale of **L** is effectively changing alpha. So larger **L** increases stability, allowing larger step sizes (e.g. fewer layers), and accelerates the spread of information (effective receptive field). 

As you will likely find, a practical downside of very large **L** is that you may be effectively smoothing the states of the network and might start to lose high frequency content. At the end of the day, with any given application,  I would recommend experimenting with a few different initializations of **L** and see how it effects performance. 

Related to your final question, *h* is the step size in pseudo-time related to viewing the forward propogation of the network as integrating an ODE, not to be confused with the step size (learning rate) used for optimization of the model weights. At the end of the day *h* effectively scales **K** and **L**, so you will need to consider *h* when you initialize **K** and **L** in regards to stability, implicit coupling, etc
